src/api/app/models/user.rb

```diff
@@ -842,11 +842,11 @@ def to_param
     to_s
   end
 
-  def nr_of_requests_that_need_work
-    Rails.cache.fetch("requests_for_#{login}", expires_in: 2.minutes) do
-      BsRequest.collection(user: login, states: %w(declined), roles: %w(creator)).count +
-      BsRequest.collection(user: login, states: %w(new), roles: %w(maintainer)).count +
-      BsRequest.collection(user: login, roles: %w(reviewer), reviewstates: %w(new), states: %w(review)).count
+  def tasks
+    Rails.cache.fetch("requests_for_#{cache_key}") do
+      declined_requests.count +
+      incoming_requests.count +
+      involved_reviews.count
```

Comment:
There is a slightly different behaviour now in this line!
Before the query also took BsRequests created by the same user into account. However, I replaced this now with the involved_reviews call which does not include BsRequests with the same creator (``.where.not(creator: login)``). I think this was a bug before and this behaviour should be the correct one.

See
https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/user.rb#L737

@adrianschroeter can you please confirm that!